---
result: FAILURE
url: http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/149/
failure_category: code
failure_cause: 'chef-client: all: chef-server-12-test'
timestamp: 2016-04-29 00:11:50 UTC
duration: 1h21m56s
triggered_by: ryancragun
active_duration: 1h21m43s
parameters:
  GIT_REF: ryan/veil
  EXPIRE_CACHE: false
change:
  git_remote: git@github.com:chef/chef-server.git
  git_commit: c8ea135c0c08df49dc85f03a26b5d5afc58a98cb
  project: chef-server
  version: 12.5.0+20160429001206
stages:
  chef-server-12-test:
    result: FAILURE
    failure_category: code
    failure_cause: 'chef-client: all'
    url: http://wilson.ci.chef.co/job/chef-server-12-test/210/
    duration: 10m14s
    runs:
      el-5:
        result: FAILURE
        failure_category: code
        failure_cause: chef-client
        failed_in:
          step: chef-client
        url: http://wilson.ci.chef.co/job/chef-server-12-test/architecture=x86_64,platform=el-5,project=chef-server,role=tester/210/
        duration: 10m14s
        logs:
          consoleText:
            396: |
              [2016-04-28T21:28:19-04:00] INFO: GET /organizations/chef/nodes/ip-10-194-15-112.us-west-2.compute.internal
              [2016-04-28T21:28:19-04:00] INFO: #<ChefZero::RestErrorResponse: 404: Object not found: chefzero://localhost:8889/nodes/ip-10-194-15-112.us-west-2.compute.internal>
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:91:in `rescue in get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:83:in `get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/endpoints/rest_object_endpoint.rb:18:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:62:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_router.rb:24:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:664:in `block in app'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `handle_socketless_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:87:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:33:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:305:in `block in send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:336:in `block in retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `loop'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:299:in `send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:144:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:111:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:604:in `load'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:588:in `find_or_create'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/policy_builder/dynamic.rb:72:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:465:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:267:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:267:in `run_with_graceful_exit_option'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:243:in `block in run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/local_mode.rb:44:in `with_server_connectivity'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:226:in `run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application/client.rb:419:in `run_application'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:58:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/bin/chef-client:26:in `<top (required)>'
              /opt/opscode/embedded/bin/chef-client:23:in `load'
              /opt/opscode/embedded/bin/chef-client:23:in `<main>'
              [2016-04-28T21:28:19-04:00] INFO:
              --- RESPONSE (404) ---
            4338: |2

                        if attempt == 120
                            raise 'ERROR: Timeout waiting for DRBD mount to return'
                        end
                        print '.'
            4413: |2
                  rescue Exception => e
                    message = <<-EOF
              ERROR:
              The /var/opt/opscode/upgrades/migration-level file is missing or corrupt!  Please read http://docs.chef.io/install_server_pre.html and correct this file before proceeding
            4727: |
              if File.exists?("/etc/opscode/chef-server.json") &&
                  !(File.exist?("/etc/opscode/private-chef.rb") || File.exist?("/etc/opscode/chef-server.rb"))
                Chef::Log.fatal("Configuration via /etc/opscode/chef-server.json is not supported. Please use /etc/opscode/chef-server.rb")
                exit!(1)
              else
            6733: |2
                      `echo 'SELECT * FROM pg_database;' | su - #{node['private_chef']['postgresql']['username']} -c '/opt/opscode/embedded/bin/psql -p #{node['private_chef']['postgresql']['port']} -U #{node['private_chef']['postgresql']['db_superuser']} postgres -t -A'`
                if $?.exitstatus != 0
                        Chef::Log.fatal("Could not connect to database, retrying in 10 seconds.")
                        sleep 10
                      else
            6742: |2

                    unless connectable
                      Chef::Log.fatal <<-ERR

              Could not connect to the postgresql database.
            7276: |2

                  if !connected
                    Chef::Log.fatal("HAProxy status socket never appeared properly!")
                    Chef::Log.fatal("See /var/log/opscode/haproxy/current for more information")
                    Kernel.exit! 1
                  end
            7327: |2

                  if !stable
                    Chef::Log.fatal("HAProxy still showing multiple active backends")
                    Chef::Log.fatal("Please check /var/log/opscode/haproxy/current locally for problems.")
                    Chef::Log.fatal("Please check your backend cluster's status for problems.")
                    Kernel.exit! 1
                  end
            8261: |2
                    # use of exit! prevents exit handlers from running, ensuring the last thing
                    # the customer sees is the descriptive error we've provided.
                    Chef::Log.fatal("\n\n#{LINE_SEP}\n#{e.message}#{LINE_SEP}")
                    exit! 128
                  end
            8444: |2
                      elsif PrivateChef["backend_vips"]["ipaddress"] =~ /\:/
                        # IPv6 addresses must have the mask
                        Chef::Log.fatal("backend_vip ipaddress field appears to be a IPv6 address without a netmask  (e.g /64, /48)")
                        exit 66
                      else
            8712: |2
                        Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
                      elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
                        Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
                                        "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
                                        "to complete the setup")
            8864: |2
                  def assert_server_config(node_name)
                    unless PrivateChef["servers"].key?(node_name)
                      Chef::Log.fatal <<-EOF
              No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
              Server configuration exists for the following hostnames:
            8891: |2
                        gen_api_fqdn
                      else
                        Chef::Log.fatal("I do not understand topology #{PrivateChef.topology} - try standalone, manual, ha, or tier.")
                        exit 55
                      end
            19372: |2

                if [ ! -d "$OCF_ROOT" ]; then
                  ha_log "ERROR: OCF_ROOT points to non-directory $OCF_ROOT."
                  exit $OCF_ERR_GENERIC
                fi
            19392: |2

                if [ -z "$OCF_RESOURCE_INSTANCE" ]; then
                  ha_log "ERROR: Need to tell us our resource instance name."
                  exit $OCF_ERR_ARGS
                fi
            19429: |2
                        loglevel=notice
                        case "${*}" in
                          *ERROR*)    loglevel=err;;
                          *WARN*)   loglevel=warning;;
                          *INFO*|info)  loglevel=info;;
            19528: |2
                      case "${__OCF_PRIO}" in
                        crit) __OCF_PRIO="CRIT";;
                        err)  __OCF_PRIO="ERROR";;
                        warn) __OCF_PRIO="WARNING";;
                        info) __OCF_PRIO="INFO";;
            20553: |2

              #------------------------------------------------------------------------------
              # ERROR REPORTING AND LOGGING
              #------------------------------------------------------------------------------
            20624: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic
            20638: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic (effectively off)
            20827: |2

              #------------------------------------------------------------------------------
              # ERROR HANDLING
              #------------------------------------------------------------------------------
            29798: |2

                if node['fqdn'].nil?
                  Chef::Log.fatal("This node's fqdn is set to nil, so a read token cannot be issued!" \
                                  "Please change your fqdn settings.")
                end
            31668: |
              \33[0mPG::StringDataRightTruncation\33[0m
              -----------------------------\33[0m
              ERROR:  value too long for type character(32)
              \33[0m
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `exec_params'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `simple_insert'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:102:in `create_superuser_in_erchef'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:50:in `block in bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:55:in `block in with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:44:in `with_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:52:in `with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:49:in `bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:46:in `block (2 levels) in from_file'\33[0m

              \33[0mResource Declaration:\33[0m
            31779: |
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:184:in `restart_service'\33[0m

              \33[0mResource Declaration:\33[0m
            31875: |
              [2016-04-28T21:31:39-04:00] INFO: Processing execute[restart_opscode-erchef_log_service] action run (private-chef::opscode-erchef line 19)
              [2016-04-28T21:31:40-04:00] INFO: execute[restart_opscode-erchef_log_service] ran successfully
              [2016-04-28T21:31:40-04:00] ERROR: Running exception handlers
              [2016-04-28T21:31:40-04:00] ERROR: Exception handlers complete
              [2016-04-28T21:31:40-04:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
              [2016-04-28T21:31:40-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
              [2016-04-28T21:31:41-04:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
              * PG::StringDataRightTruncation occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 44) had an error: PG::StringDataRightTruncation: ERROR:  value too long for type character(32)

              * Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
            31899: |
              ============================================================

              /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/config.rb:34:in `from_argv': Configuration file '/var/opt/opscode/oc-chef-pedant/etc/pedant_config.rb' not found! (RuntimeError)
                from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:54:in `setup'
                from ./bin/oc-chef-pedant:10:in `<main>'
              Build step 'Invoke XShell command' marked build as failure
              Performing Post build task...
              Match found for : : True
            31923: |
              Recording fingerprints
              Recording test results
              ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
              hudson.AbortException: No test report files were found. Configuration error?
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:116)
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:92)
                at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
                at hudson.remoting.UserRequest.perform(UserRequest.java:121)
                at hudson.remoting.UserRequest.perform(UserRequest.java:49)
                at hudson.remoting.Request$2.run(Request.java:325)
                at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
                at java.util.concurrent.FutureTask.run(FutureTask.java:262)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
                at java.lang.Thread.run(Thread.java:745)
                at ......remote call to chef-server-rhel-5-tester-4e6602(Native Method)
                at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
                at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
                at hudson.remoting.Channel.call(Channel.java:753)
                at hudson.FilePath.act(FilePath.java:978)
                at hudson.FilePath.act(FilePath.java:967)
                at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:89)
                at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:121)
                at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:138)
                at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
                at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
                at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
                at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
                at hudson.model.Build$BuildExecution.post2(Build.java:183)
                at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
                at hudson.model.Run.execute(Run.java:1766)
                at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
                at hudson.model.ResourceController.execute(ResourceController.java:98)
                at hudson.model.Executor.run(Executor.java:374)
              Regular expression run condition: Expression=[^(?!true)$], Label=[true]
              Run condition [Regular expression match] preventing perform for step [[Editable Email Notification]]
        steps:
          total: 10m14s
          before (setup time): 4m42s
          after (cleanup time): 5m28s
      el-6:
        result: FAILURE
        failure_category: code
        failure_cause: chef-client
        failed_in:
          step: chef-client
        url: http://wilson.ci.chef.co/job/chef-server-12-test/architecture=x86_64,platform=el-6,project=chef-server,role=tester/210/
        duration: 8m25s
        logs:
          consoleText:
            381: |
              [2016-04-28T21:22:42-04:00] INFO: GET /organizations/chef/nodes/ip-10-194-9-233.us-west-2.compute.internal
              [2016-04-28T21:22:42-04:00] INFO: #<ChefZero::RestErrorResponse: 404: Object not found: chefzero://localhost:8889/nodes/ip-10-194-9-233.us-west-2.compute.internal>
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:91:in `rescue in get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:83:in `get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/endpoints/rest_object_endpoint.rb:18:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:62:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_router.rb:24:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:664:in `block in app'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `handle_socketless_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:87:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:33:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:305:in `block in send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:336:in `block in retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `loop'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:299:in `send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:144:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:111:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:604:in `load'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:588:in `find_or_create'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/policy_builder/dynamic.rb:72:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:465:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:267:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:267:in `run_with_graceful_exit_option'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:243:in `block in run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/local_mode.rb:44:in `with_server_connectivity'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:226:in `run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application/client.rb:419:in `run_application'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:58:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/bin/chef-client:26:in `<top (required)>'
              /opt/opscode/embedded/bin/chef-client:23:in `load'
              /opt/opscode/embedded/bin/chef-client:23:in `<main>'
              [2016-04-28T21:22:42-04:00] INFO:
              --- RESPONSE (404) ---
            4290: |2

                        if attempt == 120
                            raise 'ERROR: Timeout waiting for DRBD mount to return'
                        end
                        print '.'
            5455: |2
                  rescue Exception => e
                    message = <<-EOF
              ERROR:
              The /var/opt/opscode/upgrades/migration-level file is missing or corrupt!  Please read http://docs.chef.io/install_server_pre.html and correct this file before proceeding
            6138: |2

                  if !connected
                    Chef::Log.fatal("HAProxy status socket never appeared properly!")
                    Chef::Log.fatal("See /var/log/opscode/haproxy/current for more information")
                    Kernel.exit! 1
                  end
            6189: |2

                  if !stable
                    Chef::Log.fatal("HAProxy still showing multiple active backends")
                    Chef::Log.fatal("Please check /var/log/opscode/haproxy/current locally for problems.")
                    Chef::Log.fatal("Please check your backend cluster's status for problems.")
                    Kernel.exit! 1
                  end
            6467: |
              if File.exists?("/etc/opscode/chef-server.json") &&
                  !(File.exist?("/etc/opscode/private-chef.rb") || File.exist?("/etc/opscode/chef-server.rb"))
                Chef::Log.fatal("Configuration via /etc/opscode/chef-server.json is not supported. Please use /etc/opscode/chef-server.rb")
                exit!(1)
              else
            7644: |2
                      `echo 'SELECT * FROM pg_database;' | su - #{node['private_chef']['postgresql']['username']} -c '/opt/opscode/embedded/bin/psql -p #{node['private_chef']['postgresql']['port']} -U #{node['private_chef']['postgresql']['db_superuser']} postgres -t -A'`
                if $?.exitstatus != 0
                        Chef::Log.fatal("Could not connect to database, retrying in 10 seconds.")
                        sleep 10
                      else
            7653: |2

                    unless connectable
                      Chef::Log.fatal <<-ERR

              Could not connect to the postgresql database.
            8187: |2
                    # use of exit! prevents exit handlers from running, ensuring the last thing
                    # the customer sees is the descriptive error we've provided.
                    Chef::Log.fatal("\n\n#{LINE_SEP}\n#{e.message}#{LINE_SEP}")
                    exit! 128
                  end
            8444: |2
                      elsif PrivateChef["backend_vips"]["ipaddress"] =~ /\:/
                        # IPv6 addresses must have the mask
                        Chef::Log.fatal("backend_vip ipaddress field appears to be a IPv6 address without a netmask  (e.g /64, /48)")
                        exit 66
                      else
            8712: |2
                        Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
                      elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
                        Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
                                        "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
                                        "to complete the setup")
            8864: |2
                  def assert_server_config(node_name)
                    unless PrivateChef["servers"].key?(node_name)
                      Chef::Log.fatal <<-EOF
              No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
              Server configuration exists for the following hostnames:
            8891: |2
                        gen_api_fqdn
                      else
                        Chef::Log.fatal("I do not understand topology #{PrivateChef.topology} - try standalone, manual, ha, or tier.")
                        exit 55
                      end
            17993: |2

                if [ ! -d "$OCF_ROOT" ]; then
                  ha_log "ERROR: OCF_ROOT points to non-directory $OCF_ROOT."
                  exit $OCF_ERR_GENERIC
                fi
            18013: |2

                if [ -z "$OCF_RESOURCE_INSTANCE" ]; then
                  ha_log "ERROR: Need to tell us our resource instance name."
                  exit $OCF_ERR_ARGS
                fi
            18050: |2
                        loglevel=notice
                        case "${*}" in
                          *ERROR*)    loglevel=err;;
                          *WARN*)   loglevel=warning;;
                          *INFO*|info)  loglevel=info;;
            18149: |2
                      case "${__OCF_PRIO}" in
                        crit) __OCF_PRIO="CRIT";;
                        err)  __OCF_PRIO="ERROR";;
                        warn) __OCF_PRIO="WARNING";;
                        info) __OCF_PRIO="INFO";;
            19242: |2

              #------------------------------------------------------------------------------
              # ERROR REPORTING AND LOGGING
              #------------------------------------------------------------------------------
            19313: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic
            19327: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic (effectively off)
            19516: |2

              #------------------------------------------------------------------------------
              # ERROR HANDLING
              #------------------------------------------------------------------------------
            29319: |2

                if node['fqdn'].nil?
                  Chef::Log.fatal("This node's fqdn is set to nil, so a read token cannot be issued!" \
                                  "Please change your fqdn settings.")
                end
            31648: |
              \33[0mPG::StringDataRightTruncation\33[0m
              -----------------------------\33[0m
              ERROR:  value too long for type character(32)
              \33[0m
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `exec_params'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `simple_insert'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:102:in `create_superuser_in_erchef'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:50:in `block in bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:55:in `block in with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:44:in `with_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:52:in `with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:49:in `bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:46:in `block (2 levels) in from_file'\33[0m

              \33[0mResource Declaration:\33[0m
            31759: |
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:184:in `restart_service'\33[0m

              \33[0mResource Declaration:\33[0m
            31855: |
              [2016-04-28T21:26:13-04:00] INFO: Processing execute[restart_opscode-erchef_log_service] action run (private-chef::opscode-erchef line 19)
              [2016-04-28T21:26:14-04:00] INFO: execute[restart_opscode-erchef_log_service] ran successfully
              [2016-04-28T21:26:14-04:00] ERROR: Running exception handlers
              [2016-04-28T21:26:14-04:00] ERROR: Exception handlers complete
              [2016-04-28T21:26:14-04:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
              [2016-04-28T21:26:14-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
              [2016-04-28T21:26:14-04:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
              * PG::StringDataRightTruncation occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 44) had an error: PG::StringDataRightTruncation: ERROR:  value too long for type character(32)

              * Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
            31879: |
              ============================================================

              /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/config.rb:34:in `from_argv': Configuration file '/var/opt/opscode/oc-chef-pedant/etc/pedant_config.rb' not found! (RuntimeError)
                from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:54:in `setup'
                from ./bin/oc-chef-pedant:10:in `<main>'
              Build step 'Invoke XShell command' marked build as failure
              Performing Post build task...
              Match found for : : True
            31903: |
              Recording fingerprints
              Recording test results
              ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
              hudson.AbortException: No test report files were found. Configuration error?
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:116)
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:92)
                at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
                at hudson.remoting.UserRequest.perform(UserRequest.java:121)
                at hudson.remoting.UserRequest.perform(UserRequest.java:49)
                at hudson.remoting.Request$2.run(Request.java:325)
                at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
                at java.util.concurrent.FutureTask.run(FutureTask.java:262)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
                at java.lang.Thread.run(Thread.java:745)
                at ......remote call to chef-server-rhel-6-tester-8151c8(Native Method)
                at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
                at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
                at hudson.remoting.Channel.call(Channel.java:753)
                at hudson.FilePath.act(FilePath.java:978)
                at hudson.FilePath.act(FilePath.java:967)
                at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:89)
                at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:121)
                at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:138)
                at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
                at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
                at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
                at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
                at hudson.model.Build$BuildExecution.post2(Build.java:183)
                at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
                at hudson.model.Run.execute(Run.java:1766)
                at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
                at hudson.model.ResourceController.execute(ResourceController.java:98)
                at hudson.model.Executor.run(Executor.java:374)
              Regular expression run condition: Expression=[^(?!true)$], Label=[true]
              Run condition [Regular expression match] preventing perform for step [[Editable Email Notification]]
        steps:
          total: 8m25s
          before (setup time): "-56s"
          after (cleanup time): 9m16s
      el-7:
        result: FAILURE
        failure_category: code
        failure_cause: chef-client
        failed_in:
          step: chef-client
        url: http://wilson.ci.chef.co/job/chef-server-12-test/architecture=x86_64,platform=el-7,project=chef-server,role=tester/210/
        duration: 8m21s
        logs:
          consoleText:
            55841: |
              warning: file /opt/opscode/embedded/lib/ruby/gems/2.1.0/doc/nokogiri-1.6.7.2/ri/Nokogiri/XML/SyntaxError/int1-i.ri: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/ruby/gems/2.1.0/doc/nokogiri-1.6.7.2/ri/Nokogiri/XML/SyntaxError/file-i.ri: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/ruby/gems/2.1.0/doc/nokogiri-1.6.7.2/ri/Nokogiri/XML/SyntaxError/fatal%3f-i.ri: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/ruby/gems/2.1.0/doc/nokogiri-1.6.7.2/ri/Nokogiri/XML/SyntaxError/error%3f-i.ri: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/ruby/gems/2.1.0/doc/nokogiri-1.6.7.2/ri/Nokogiri/XML/SyntaxError/domain-i.ri: remove failed: No such file or directory
            63054: |
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/auto/Test/File/.packlist: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/auto/Test/File: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/auto/Test/Fatal/.packlist: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/auto/Test/Fatal: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/auto/Test/Exception/.packlist: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/auto/Test/Exception: remove failed: No such file or directory
            64000: |
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-File-1.44/MYMETA.json: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-File-1.44: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-Fatal-0.014/install.json: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-Fatal-0.014/MYMETA.json: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-Fatal-0.014: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-Exception-0.43/install.json: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/x86_64-linux-thread-multi/.meta/Test-Exception-0.43/MYMETA.json: remove failed: No such file or directory
            64382: |
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/Test/File.pm: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/Test/File: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/Test/Fatal.pm: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/Test/Exception.pm: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/site_perl/5.18.1/Test/Dir/Base.pm: remove failed: No such file or directory
            68210: |
              warning: file /opt/opscode/embedded/lib/perl5/5.18.1/File/Basename.pm: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/5.18.1/File: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/5.18.1/Fatal.pm: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/5.18.1/ExtUtils/xsubpp: remove failed: No such file or directory
              warning: file /opt/opscode/embedded/lib/perl5/5.18.1/ExtUtils/typemap: remove failed: No such file or directory
            77667: |
              [2016-04-28T21:27:34-04:00] INFO: GET /organizations/chef/nodes/ip-10-194-10-226.us-west-2.compute.internal
              [2016-04-28T21:27:34-04:00] INFO: #<ChefZero::RestErrorResponse: 404: Object not found: chefzero://localhost:8889/nodes/ip-10-194-10-226.us-west-2.compute.internal>
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:91:in `rescue in get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:83:in `get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/endpoints/rest_object_endpoint.rb:18:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:62:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_router.rb:24:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:664:in `block in app'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `handle_socketless_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:87:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:33:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:305:in `block in send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:336:in `block in retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `loop'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:299:in `send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:144:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:111:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:604:in `load'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:588:in `find_or_create'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/policy_builder/dynamic.rb:72:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:465:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:267:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:267:in `run_with_graceful_exit_option'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:243:in `block in run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/local_mode.rb:44:in `with_server_connectivity'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:226:in `run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application/client.rb:419:in `run_application'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:58:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/bin/chef-client:26:in `<top (required)>'
              /opt/opscode/embedded/bin/chef-client:23:in `load'
              /opt/opscode/embedded/bin/chef-client:23:in `<main>'
              [2016-04-28T21:27:34-04:00] INFO:
              --- RESPONSE (404) ---
            81031: |2

                        if attempt == 120
                            raise 'ERROR: Timeout waiting for DRBD mount to return'
                        end
                        print '.'
            81909: |
              if File.exists?("/etc/opscode/chef-server.json") &&
                  !(File.exist?("/etc/opscode/private-chef.rb") || File.exist?("/etc/opscode/chef-server.rb"))
                Chef::Log.fatal("Configuration via /etc/opscode/chef-server.json is not supported. Please use /etc/opscode/chef-server.rb")
                exit!(1)
              else
            82631: |2

                  if !connected
                    Chef::Log.fatal("HAProxy status socket never appeared properly!")
                    Chef::Log.fatal("See /var/log/opscode/haproxy/current for more information")
                    Kernel.exit! 1
                  end
            82682: |2

                  if !stable
                    Chef::Log.fatal("HAProxy still showing multiple active backends")
                    Chef::Log.fatal("Please check /var/log/opscode/haproxy/current locally for problems.")
                    Chef::Log.fatal("Please check your backend cluster's status for problems.")
                    Kernel.exit! 1
                  end
            83755: |2
                  rescue Exception => e
                    message = <<-EOF
              ERROR:
              The /var/opt/opscode/upgrades/migration-level file is missing or corrupt!  Please read http://docs.chef.io/install_server_pre.html and correct this file before proceeding
            84654: |2
                      `echo 'SELECT * FROM pg_database;' | su - #{node['private_chef']['postgresql']['username']} -c '/opt/opscode/embedded/bin/psql -p #{node['private_chef']['postgresql']['port']} -U #{node['private_chef']['postgresql']['db_superuser']} postgres -t -A'`
                if $?.exitstatus != 0
                        Chef::Log.fatal("Could not connect to database, retrying in 10 seconds.")
                        sleep 10
                      else
            84663: |2

                    unless connectable
                      Chef::Log.fatal <<-ERR

              Could not connect to the postgresql database.
            86300: |2
                      elsif PrivateChef["backend_vips"]["ipaddress"] =~ /\:/
                        # IPv6 addresses must have the mask
                        Chef::Log.fatal("backend_vip ipaddress field appears to be a IPv6 address without a netmask  (e.g /64, /48)")
                        exit 66
                      else
            86568: |2
                        Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
                      elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
                        Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
                                        "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
                                        "to complete the setup")
            86720: |2
                  def assert_server_config(node_name)
                    unless PrivateChef["servers"].key?(node_name)
                      Chef::Log.fatal <<-EOF
              No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
              Server configuration exists for the following hostnames:
            86747: |2
                        gen_api_fqdn
                      else
                        Chef::Log.fatal("I do not understand topology #{PrivateChef.topology} - try standalone, manual, ha, or tier.")
                        exit 55
                      end
            86973: |2
                    # use of exit! prevents exit handlers from running, ensuring the last thing
                    # the customer sees is the descriptive error we've provided.
                    Chef::Log.fatal("\n\n#{LINE_SEP}\n#{e.message}#{LINE_SEP}")
                    exit! 128
                  end
            92892: |2

                if [ ! -d "$OCF_ROOT" ]; then
                  ha_log "ERROR: OCF_ROOT points to non-directory $OCF_ROOT."
                  exit $OCF_ERR_GENERIC
                fi
            92912: |2

                if [ -z "$OCF_RESOURCE_INSTANCE" ]; then
                  ha_log "ERROR: Need to tell us our resource instance name."
                  exit $OCF_ERR_ARGS
                fi
            92949: |2
                        loglevel=notice
                        case "${*}" in
                          *ERROR*)    loglevel=err;;
                          *WARN*)   loglevel=warning;;
                          *INFO*|info)  loglevel=info;;
            93048: |2
                      case "${__OCF_PRIO}" in
                        crit) __OCF_PRIO="CRIT";;
                        err)  __OCF_PRIO="ERROR";;
                        warn) __OCF_PRIO="WARNING";;
                        info) __OCF_PRIO="INFO";;
            94081: |2

              #------------------------------------------------------------------------------
              # ERROR REPORTING AND LOGGING
              #------------------------------------------------------------------------------
            94152: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic
            94166: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic (effectively off)
            94355: |2

              #------------------------------------------------------------------------------
              # ERROR HANDLING
              #------------------------------------------------------------------------------
            107479: |2

                if node['fqdn'].nil?
                  Chef::Log.fatal("This node's fqdn is set to nil, so a read token cannot be issued!" \
                                  "Please change your fqdn settings.")
                end
            108906: |
              \33[0mPG::StringDataRightTruncation\33[0m
              -----------------------------\33[0m
              ERROR:  value too long for type character(32)
              \33[0m
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `exec_params'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `simple_insert'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:102:in `create_superuser_in_erchef'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:50:in `block in bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:55:in `block in with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:44:in `with_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:52:in `with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:49:in `bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:46:in `block (2 levels) in from_file'\33[0m

              \33[0mResource Declaration:\33[0m
            109017: |
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:184:in `restart_service'\33[0m

              \33[0mResource Declaration:\33[0m
            109113: |
              [2016-04-28T21:29:48-04:00] INFO: Processing execute[restart_opscode-erchef_log_service] action run (private-chef::opscode-erchef line 19)
              [2016-04-28T21:29:48-04:00] INFO: execute[restart_opscode-erchef_log_service] ran successfully
              [2016-04-28T21:29:48-04:00] ERROR: Running exception handlers
              [2016-04-28T21:29:48-04:00] ERROR: Exception handlers complete
              [2016-04-28T21:29:48-04:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
              [2016-04-28T21:29:48-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
              [2016-04-28T21:29:49-04:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
              * PG::StringDataRightTruncation occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 44) had an error: PG::StringDataRightTruncation: ERROR:  value too long for type character(32)

              * Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
            109137: |
              ============================================================

              /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/config.rb:34:in `from_argv': Configuration file '/var/opt/opscode/oc-chef-pedant/etc/pedant_config.rb' not found! (RuntimeError)
                from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:54:in `setup'
                from ./bin/oc-chef-pedant:10:in `<main>'
              Build step 'Invoke XShell command' marked build as failure
              Performing Post build task...
              Match found for : : True
            109161: |
              Recording fingerprints
              Recording test results
              ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
              hudson.AbortException: No test report files were found. Configuration error?
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:116)
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:92)
                at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
                at hudson.remoting.UserRequest.perform(UserRequest.java:121)
                at hudson.remoting.UserRequest.perform(UserRequest.java:49)
                at hudson.remoting.Request$2.run(Request.java:325)
                at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
                at java.util.concurrent.FutureTask.run(FutureTask.java:262)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
                at java.lang.Thread.run(Thread.java:745)
                at ......remote call to chef-server-rhel-7-tester-d586ce(Native Method)
                at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
                at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
                at hudson.remoting.Channel.call(Channel.java:753)
                at hudson.FilePath.act(FilePath.java:978)
                at hudson.FilePath.act(FilePath.java:967)
                at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:89)
                at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:121)
                at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:138)
                at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
                at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
                at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
                at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
                at hudson.model.Build$BuildExecution.post2(Build.java:183)
                at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
                at hudson.model.Run.execute(Run.java:1766)
                at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
                at hudson.model.ResourceController.execute(ResourceController.java:98)
                at hudson.model.Executor.run(Executor.java:374)
              Regular expression run condition: Expression=[^(?!true)$], Label=[true]
              Run condition [Regular expression match] preventing perform for step [[Editable Email Notification]]
        steps:
          total: 8m21s
          before (setup time): 3m57s
          after (cleanup time): 4m20s
      ubuntu-10.04:
        result: FAILURE
        failure_category: code
        failure_cause: chef-client
        failed_in:
          step: chef-client
        url: http://wilson.ci.chef.co/job/chef-server-12-test/architecture=x86_64,platform=ubuntu-10.04,project=chef-server,role=tester/210/
        duration: 7m17s
        logs:
          consoleText:
            409: |
              [2016-04-29T01:26:05+00:00] INFO: GET /organizations/chef/nodes/ip-10-194-13-26.us-west-2.compute.internal
              [2016-04-29T01:26:05+00:00] INFO: #<ChefZero::RestErrorResponse: 404: Object not found: chefzero://localhost:8889/nodes/ip-10-194-13-26.us-west-2.compute.internal>
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:91:in `rescue in get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:83:in `get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/endpoints/rest_object_endpoint.rb:18:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:62:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_router.rb:24:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:664:in `block in app'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `handle_socketless_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:87:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:33:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:305:in `block in send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:336:in `block in retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `loop'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:299:in `send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:144:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:111:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:604:in `load'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:588:in `find_or_create'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/policy_builder/dynamic.rb:72:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:465:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:267:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:267:in `run_with_graceful_exit_option'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:243:in `block in run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/local_mode.rb:44:in `with_server_connectivity'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:226:in `run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application/client.rb:419:in `run_application'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:58:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/bin/chef-client:26:in `<top (required)>'
              /opt/opscode/embedded/bin/chef-client:23:in `load'
              /opt/opscode/embedded/bin/chef-client:23:in `<main>'
              [2016-04-29T01:26:05+00:00] INFO:
              --- RESPONSE (404) ---
            4172: |2

                        if attempt == 120
                            raise 'ERROR: Timeout waiting for DRBD mount to return'
                        end
                        print '.'
            4724: |
              if File.exists?("/etc/opscode/chef-server.json") &&
                  !(File.exist?("/etc/opscode/private-chef.rb") || File.exist?("/etc/opscode/chef-server.rb"))
                Chef::Log.fatal("Configuration via /etc/opscode/chef-server.json is not supported. Please use /etc/opscode/chef-server.rb")
                exit!(1)
              else
            5824: |2
                  rescue Exception => e
                    message = <<-EOF
              ERROR:
              The /var/opt/opscode/upgrades/migration-level file is missing or corrupt!  Please read http://docs.chef.io/install_server_pre.html and correct this file before proceeding
            6365: |2
                [2016-04-29T01:26:07+00:00] INFO: Storing updated cookbooks/private-chef/recipes/oc-chef-pedant.rb in the cache.
                if !connected
                    Chef::Log.fatal("HAProxy status socket never appeared properly!")
                    Chef::Log.fatal("See /var/log/opscode/haproxy/current for more information")
                    Kernel.exit! 1
                  end
            6416: |2

                  if !stable
                    Chef::Log.fatal("HAProxy still showing multiple active backends")
                    Chef::Log.fatal("Please check /var/log/opscode/haproxy/current locally for problems.")
                    Chef::Log.fatal("Please check your backend cluster's status for problems.")
                    Kernel.exit! 1
                  end
            7231: |2
                      `echo 'SELECT * FROM pg_database;' | su - #{node['private_chef']['postgresql']['username']} -c '/opt/opscode/embedded/bin/psql -p #{node['private_chef']['postgresql']['port']} -U #{node['private_chef']['postgresql']['db_superuser']} postgres -t -A'`
                if $?.exitstatus != 0
                        Chef::Log.fatal("Could not connect to database, retrying in 10 seconds.")
                        sleep 10
                      else
            7240: |2

                    unless connectable
                      Chef::Log.fatal <<-ERR

              Could not connect to the postgresql database.
            7648: |2
                    # use of exit! prevents exit handlers from running, ensuring the last thing
                    # the customer sees is the descriptive error we've provided.
                    Chef::Log.fatal("\n\n#{LINE_SEP}\n#{e.message}#{LINE_SEP}")
                    exit! 128
                  end
            8566: |2
                      elsif PrivateChef["backend_vips"]["ipaddress"] =~ /\:/
                        # IPv6 addresses must have the mask
                        Chef::Log.fatal("backend_vip ipaddress field appears to be a IPv6 address without a netmask  (e.g /64, /48)")
                        exit 66
                      else
            8834: |2
                        Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
                      elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
                        Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
                                        "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
                                        "to complete the setup")
            8986: |2
                  def assert_server_config(node_name)
                    unless PrivateChef["servers"].key?(node_name)
                      Chef::Log.fatal <<-EOF
              No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
              Server configuration exists for the following hostnames:
            9013: |2
                        gen_api_fqdn
                      else
                        Chef::Log.fatal("I do not understand topology #{PrivateChef.topology} - try standalone, manual, ha, or tier.")
                        exit 55
                      end
            13556: |2

              #------------------------------------------------------------------------------
              # ERROR REPORTING AND LOGGING
              #------------------------------------------------------------------------------
            13627: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic
            13641: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic (effectively off)
            13830: |2

              #------------------------------------------------------------------------------
              # ERROR HANDLING
              #------------------------------------------------------------------------------
            17002: |2

                if [ ! -d "$OCF_ROOT" ]; then
                  ha_log "ERROR: OCF_ROOT points to non-directory $OCF_ROOT."
                  exit $OCF_ERR_GENERIC
                fi
            17022: |2

                if [ -z "$OCF_RESOURCE_INSTANCE" ]; then
                  ha_log "ERROR: Need to tell us our resource instance name."
                  exit $OCF_ERR_ARGS
                fi
            17059: |2
                        loglevel=notice
                        case "${*}" in
                          *ERROR*)    loglevel=err;;
                          *WARN*)   loglevel=warning;;
                          *INFO*|info)  loglevel=info;;
            17158: |2
                      case "${__OCF_PRIO}" in
                        crit) __OCF_PRIO="CRIT";;
                        err)  __OCF_PRIO="ERROR";;
                        warn) __OCF_PRIO="WARNING";;
                        info) __OCF_PRIO="INFO";;
            29448: |2

                if node['fqdn'].nil?
                  Chef::Log.fatal("This node's fqdn is set to nil, so a read token cannot be issued!" \
                                  "Please change your fqdn settings.")
                end
            31676: |
              \33[0mPG::StringDataRightTruncation\33[0m
              -----------------------------\33[0m
              ERROR:  value too long for type character(32)
              \33[0m
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `exec_params'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `simple_insert'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:102:in `create_superuser_in_erchef'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:50:in `block in bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:55:in `block in with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:44:in `with_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:52:in `with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:49:in `bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:46:in `block (2 levels) in from_file'\33[0m

              \33[0mResource Declaration:\33[0m
            31787: |
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:184:in `restart_service'\33[0m

              \33[0mResource Declaration:\33[0m
            31883: |
              [2016-04-29T01:28:42+00:00] INFO: Processing execute[restart_opscode-erchef_log_service] action run (private-chef::opscode-erchef line 19)
              [2016-04-29T01:28:43+00:00] INFO: execute[restart_opscode-erchef_log_service] ran successfully
              [2016-04-29T01:28:43+00:00] ERROR: Running exception handlers
              [2016-04-29T01:28:43+00:00] ERROR: Exception handlers complete
              [2016-04-29T01:28:43+00:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
              [2016-04-29T01:28:43+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
              [2016-04-29T01:28:43+00:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
              * PG::StringDataRightTruncation occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 44) had an error: PG::StringDataRightTruncation: ERROR:  value too long for type character(32)

              * Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
            31907: |
              ============================================================

              /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/config.rb:34:in `from_argv': Configuration file '/var/opt/opscode/oc-chef-pedant/etc/pedant_config.rb' not found! (RuntimeError)
                from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:54:in `setup'
                from ./bin/oc-chef-pedant:10:in `<main>'
              Build step 'Invoke XShell command' marked build as failure
              Performing Post build task...
              Match found for : : True
            31931: |
              Recording fingerprints
              Recording test results
              ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
              hudson.AbortException: No test report files were found. Configuration error?
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:116)
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:92)
                at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
                at hudson.remoting.UserRequest.perform(UserRequest.java:121)
                at hudson.remoting.UserRequest.perform(UserRequest.java:49)
                at hudson.remoting.Request$2.run(Request.java:325)
                at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
                at java.util.concurrent.FutureTask.run(FutureTask.java:262)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
                at java.lang.Thread.run(Thread.java:745)
                at ......remote call to chef-server-ubuntu-10.04-tester-f80d1f(Native Method)
                at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
                at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
                at hudson.remoting.Channel.call(Channel.java:753)
                at hudson.FilePath.act(FilePath.java:978)
                at hudson.FilePath.act(FilePath.java:967)
                at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:89)
                at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:121)
                at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:138)
                at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
                at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
                at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
                at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
                at hudson.model.Build$BuildExecution.post2(Build.java:183)
                at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
                at hudson.model.Run.execute(Run.java:1766)
                at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
                at hudson.model.ResourceController.execute(ResourceController.java:98)
                at hudson.model.Executor.run(Executor.java:374)
              Regular expression run condition: Expression=[^(?!true)$], Label=[true]
              Run condition [Regular expression match] preventing perform for step [[Editable Email Notification]]
        steps:
          total: 7m17s
          before (setup time): 2m28s
          after (cleanup time): 4m46s
      ubuntu-12.04:
        result: FAILURE
        failure_category: code
        failure_cause: chef-client
        failed_in:
          step: chef-client
        url: http://wilson.ci.chef.co/job/chef-server-12-test/architecture=x86_64,platform=ubuntu-12.04,project=chef-server,role=tester/210/
        duration: 7m30s
        logs:
          consoleText:
            407: |
              [2016-04-29T01:26:34+00:00] INFO: GET /organizations/chef/nodes/ip-10-194-11-179.us-west-2.compute.internal
              [2016-04-29T01:26:34+00:00] INFO: #<ChefZero::RestErrorResponse: 404: Object not found: chefzero://localhost:8889/nodes/ip-10-194-11-179.us-west-2.compute.internal>
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:91:in `rescue in get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:83:in `get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/endpoints/rest_object_endpoint.rb:18:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:62:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_router.rb:24:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:664:in `block in app'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `handle_socketless_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:87:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:33:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:305:in `block in send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:336:in `block in retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `loop'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:299:in `send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:144:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:111:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:604:in `load'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:588:in `find_or_create'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/policy_builder/dynamic.rb:72:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:465:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:267:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:267:in `run_with_graceful_exit_option'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:243:in `block in run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/local_mode.rb:44:in `with_server_connectivity'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:226:in `run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application/client.rb:419:in `run_application'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:58:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/bin/chef-client:26:in `<top (required)>'
              /opt/opscode/embedded/bin/chef-client:23:in `load'
              /opt/opscode/embedded/bin/chef-client:23:in `<main>'
              [2016-04-29T01:26:34+00:00] INFO:
              --- RESPONSE (404) ---
            4201: |2

                        if attempt == 120
                            raise 'ERROR: Timeout waiting for DRBD mount to return'
                        end
                        print '.'
            5488: |2
                      `echo 'SELECT * FROM pg_database;' | su - #{node['private_chef']['postgresql']['username']} -c '/opt/opscode/embedded/bin/psql -p #{node['private_chef']['postgresql']['port']} -U #{node['private_chef']['postgresql']['db_superuser']} postgres -t -A'`
                if $?.exitstatus != 0
                        Chef::Log.fatal("Could not connect to database, retrying in 10 seconds.")
                        sleep 10
                      else
            5497: |2

                    unless connectable
                      Chef::Log.fatal <<-ERR

              Could not connect to the postgresql database.
            6096: |2
                  rescue Exception => e
                    message = <<-EOF
              ERROR:
              The /var/opt/opscode/upgrades/migration-level file is missing or corrupt!  Please read http://docs.chef.io/install_server_pre.html and correct this file before proceeding
            6236: |
              if File.exists?("/etc/opscode/chef-server.json") &&
                  !(File.exist?("/etc/opscode/private-chef.rb") || File.exist?("/etc/opscode/chef-server.rb"))
                Chef::Log.fatal("Configuration via /etc/opscode/chef-server.json is not supported. Please use /etc/opscode/chef-server.rb")
                exit!(1)
              else
            8373: |2
                      elsif PrivateChef["backend_vips"]["ipaddress"] =~ /\:/
                        # IPv6 addresses must have the mask
                        Chef::Log.fatal("backend_vip ipaddress field appears to be a IPv6 address without a netmask  (e.g /64, /48)")
                        exit 66
                      else
            8642: |2
                        Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
                      elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
                        Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
                                        "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
                                        "to complete the setup")
            8794: |2
                  def assert_server_config(node_name)
                    unless PrivateChef["servers"].key?(node_name)
                      Chef::Log.fatal <<-EOF
              No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
              Server configuration exists for the following hostnames:
            8821: |2
                        gen_api_fqdn
                      else
                        Chef::Log.fatal("I do not understand topology #{PrivateChef.topology} - try standalone, manual, ha, or tier.")
                        exit 55
                      end
            8996: |2

                  if !connected
                    Chef::Log.fatal("HAProxy status socket never appeared properly!")
                    Chef::Log.fatal("See /var/log/opscode/haproxy/current for more information")
                    Kernel.exit! 1
                  end
            9047: |2

                  if !stable
                    Chef::Log.fatal("HAProxy still showing multiple active backends")
                    Chef::Log.fatal("Please check /var/log/opscode/haproxy/current locally for problems.")
                    Chef::Log.fatal("Please check your backend cluster's status for problems.")
                    Kernel.exit! 1
                  end
            9306: |2
                    # use of exit! prevents exit handlers from running, ensuring the last thing
                    # the customer sees is the descriptive error we've provided.
                    Chef::Log.fatal("\n\n#{LINE_SEP}\n#{e.message}#{LINE_SEP}")
                    exit! 128
                  end
            18605: |2

              #------------------------------------------------------------------------------
              # ERROR REPORTING AND LOGGING
              #------------------------------------------------------------------------------
            18676: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic
            18690: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic (effectively off)
            18879: |2

              #------------------------------------------------------------------------------
              # ERROR HANDLING
              #------------------------------------------------------------------------------
            20164: |2

                if [ ! -d "$OCF_ROOT" ]; then
                  ha_log "ERROR: OCF_ROOT points to non-directory $OCF_ROOT."
                  exit $OCF_ERR_GENERIC
                fi
            20184: |2

                if [ -z "$OCF_RESOURCE_INSTANCE" ]; then
                  ha_log "ERROR: Need to tell us our resource instance name."
                  exit $OCF_ERR_ARGS
                fi
            20221: |2
                        loglevel=notice
                        case "${*}" in
                          *ERROR*)    loglevel=err;;
                          *WARN*)   loglevel=warning;;
                          *INFO*|info)  loglevel=info;;
            20320: |2
                      case "${__OCF_PRIO}" in
                        crit) __OCF_PRIO="CRIT";;
                        err)  __OCF_PRIO="ERROR";;
                        warn) __OCF_PRIO="WARNING";;
                        info) __OCF_PRIO="INFO";;
            30108: |2

                if node['fqdn'].nil?
                  Chef::Log.fatal("This node's fqdn is set to nil, so a read token cannot be issued!" \
                                  "Please change your fqdn settings.")
                end
            31673: |
              \33[0mPG::StringDataRightTruncation\33[0m
              -----------------------------\33[0m
              ERROR:  value too long for type character(32)
              \33[0m
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `exec_params'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `simple_insert'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:102:in `create_superuser_in_erchef'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:50:in `block in bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:55:in `block in with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:44:in `with_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:52:in `with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:49:in `bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:46:in `block (2 levels) in from_file'\33[0m

              \33[0mResource Declaration:\33[0m
            31784: |
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:184:in `restart_service'\33[0m

              \33[0mResource Declaration:\33[0m
            31880: |
              [2016-04-29T01:28:57+00:00] INFO: Processing execute[restart_opscode-erchef_log_service] action run (private-chef::opscode-erchef line 19)
              [2016-04-29T01:28:57+00:00] INFO: execute[restart_opscode-erchef_log_service] ran successfully
              [2016-04-29T01:28:57+00:00] ERROR: Running exception handlers
              [2016-04-29T01:28:57+00:00] ERROR: Exception handlers complete
              [2016-04-29T01:28:57+00:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
              [2016-04-29T01:28:57+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
              [2016-04-29T01:28:58+00:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
              * PG::StringDataRightTruncation occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 44) had an error: PG::StringDataRightTruncation: ERROR:  value too long for type character(32)

              * Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
            31904: |
              ============================================================

              /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/config.rb:34:in `from_argv': Configuration file '/var/opt/opscode/oc-chef-pedant/etc/pedant_config.rb' not found! (RuntimeError)
                from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:54:in `setup'
                from ./bin/oc-chef-pedant:10:in `<main>'
              Build step 'Invoke XShell command' marked build as failure
              Performing Post build task...
              Match found for : : True
            31928: |
              Recording fingerprints
              Recording test results
              ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
              hudson.AbortException: No test report files were found. Configuration error?
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:116)
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:92)
                at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
                at hudson.remoting.UserRequest.perform(UserRequest.java:121)
                at hudson.remoting.UserRequest.perform(UserRequest.java:49)
                at hudson.remoting.Request$2.run(Request.java:325)
                at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
                at java.util.concurrent.FutureTask.run(FutureTask.java:262)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
                at java.lang.Thread.run(Thread.java:745)
                at ......remote call to chef-server-ubuntu-12.04-tester-28c735(Native Method)
                at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
                at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
                at hudson.remoting.Channel.call(Channel.java:753)
                at hudson.FilePath.act(FilePath.java:978)
                at hudson.FilePath.act(FilePath.java:967)
                at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:89)
                at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:121)
                at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:138)
                at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
                at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
                at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
                at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
                at hudson.model.Build$BuildExecution.post2(Build.java:183)
                at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
                at hudson.model.Run.execute(Run.java:1766)
                at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
                at hudson.model.ResourceController.execute(ResourceController.java:98)
                at hudson.model.Executor.run(Executor.java:374)
              Regular expression run condition: Expression=[^(?!true)$], Label=[true]
              Run condition [Regular expression match] preventing perform for step [[Editable Email Notification]]
        steps:
          total: 7m30s
          before (setup time): 2m58s
          after (cleanup time): 4m30s
      ubuntu-14.04:
        result: FAILURE
        failure_category: code
        failure_cause: chef-client
        failed_in:
          step: chef-client
        url: http://wilson.ci.chef.co/job/chef-server-12-test/architecture=x86_64,platform=ubuntu-14.04,project=chef-server,role=tester/210/
        duration: 7m6s
        logs:
          consoleText:
            437: |
              [2016-04-29T01:26:00+00:00] INFO: GET /organizations/chef/nodes/ip-10-194-13-226.us-west-2.compute.internal
              [2016-04-29T01:26:00+00:00] INFO: #<ChefZero::RestErrorResponse: 404: Object not found: chefzero://localhost:8889/nodes/ip-10-194-13-226.us-west-2.compute.internal>
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:91:in `rescue in get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:83:in `get_data'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/endpoints/rest_object_endpoint.rb:18:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_base.rb:62:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/rest_router.rb:24:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:664:in `block in app'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `call'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/server.rb:336:in `handle_socketless_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:87:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-zero-4.6.1/lib/chef_zero/socketless_server_map.rb:33:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http/socketless_chef_zero_client.rb:154:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:305:in `block in send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:336:in `block in retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `loop'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:334:in `retrying_http_errors'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:299:in `send_http_request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:144:in `request'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/http.rb:111:in `get'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:604:in `load'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/node.rb:588:in `find_or_create'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/policy_builder/dynamic.rb:72:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:465:in `load_node'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/client.rb:267:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:267:in `run_with_graceful_exit_option'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:243:in `block in run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/local_mode.rb:44:in `with_server_connectivity'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:226:in `run_chef_client'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application/client.rb:419:in `run_application'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/lib/chef/application.rb:58:in `run'
              /opt/opscode/embedded/lib/ruby/gems/2.1.0/gems/chef-12.10.25/bin/chef-client:26:in `<top (required)>'
              /opt/opscode/embedded/bin/chef-client:23:in `load'
              /opt/opscode/embedded/bin/chef-client:23:in `<main>'
              [2016-04-29T01:26:00+00:00] INFO:
              --- RESPONSE (404) ---
            4310: |2

                        if attempt == 120
                            raise 'ERROR: Timeout waiting for DRBD mount to return'
                        end
                        print '.'
            4857: |2

                  if !connected
                    Chef::Log.fatal("HAProxy status socket never appeared properly!")
                    Chef::Log.fatal("See /var/log/opscode/haproxy/current for more information")
                    Kernel.exit! 1
                  end
            4908: |2

                  if !stable
                    Chef::Log.fatal("HAProxy still showing multiple active backends")
                    Chef::Log.fatal("Please check /var/log/opscode/haproxy/current locally for problems.")
                    Chef::Log.fatal("Please check your backend cluster's status for problems.")
                    Kernel.exit! 1
                  end
            6086: |
              if File.exists?("/etc/opscode/chef-server.json") &&
                  !(File.exist?("/etc/opscode/private-chef.rb") || File.exist?("/etc/opscode/chef-server.rb"))
                Chef::Log.fatal("Configuration via /etc/opscode/chef-server.json is not supported. Please use /etc/opscode/chef-server.rb")
                exit!(1)
              else
            7157: |2
                  rescue Exception => e
                    message = <<-EOF
              ERROR:
              The /var/opt/opscode/upgrades/migration-level file is missing or corrupt!  Please read http://docs.chef.io/install_server_pre.html and correct this file before proceeding
            7321: |2
                      `echo 'SELECT * FROM pg_database;' | su - #{node['private_chef']['postgresql']['username']} -c '/opt/opscode/embedded/bin/psql -p #{node['private_chef']['postgresql']['port']} -U #{node['private_chef']['postgresql']['db_superuser']} postgres -t -A'`
                if $?.exitstatus != 0
                        Chef::Log.fatal("Could not connect to database, retrying in 10 seconds.")
                        sleep 10
                      else
            7330: |2

                    unless connectable
                      Chef::Log.fatal <<-ERR

              Could not connect to the postgresql database.
            9679: |2
                      elsif PrivateChef["backend_vips"]["ipaddress"] =~ /\:/
                        # IPv6 addresses must have the mask
                        Chef::Log.fatal("backend_vip ipaddress field appears to be a IPv6 address without a netmask  (e.g /64, /48)")
                        exit 66
                      else
            9947: |2
                        Veil::CredentialCollection::ChefSecretsFile.from_file(secrets_json)
                      elsif PrivateChef["topology"] == "ha" && !PrivateChef["servers"][node_name]["bootstrap"]
                        Chef::Log.fatal("In an H/A topology the secrets must be created on the bootstrap node. "\
                                        "Please copy the contents of /etc/opscode/ from your bootstrap Server " \
                                        "to complete the setup")
            10099: |2
                  def assert_server_config(node_name)
                    unless PrivateChef["servers"].key?(node_name)
                      Chef::Log.fatal <<-EOF
              No server configuration found for "#{node_name}" in /etc/opscode/chef-server.rb.
              Server configuration exists for the following hostnames:
            10126: |2
                        gen_api_fqdn
                      else
                        Chef::Log.fatal("I do not understand topology #{PrivateChef.topology} - try standalone, manual, ha, or tier.")
                        exit 55
                      end
            10332: |2
                    # use of exit! prevents exit handlers from running, ensuring the last thing
                    # the customer sees is the descriptive error we've provided.
                    Chef::Log.fatal("\n\n#{LINE_SEP}\n#{e.message}#{LINE_SEP}")
                    exit! 128
                  end
            13857: |2

                if [ ! -d "$OCF_ROOT" ]; then
                  ha_log "ERROR: OCF_ROOT points to non-directory $OCF_ROOT."
                  exit $OCF_ERR_GENERIC
                fi
            13877: |2

                if [ -z "$OCF_RESOURCE_INSTANCE" ]; then
                  ha_log "ERROR: Need to tell us our resource instance name."
                  exit $OCF_ERR_ARGS
                fi
            13914: |2
                        loglevel=notice
                        case "${*}" in
                          *ERROR*)    loglevel=err;;
                          *WARN*)   loglevel=warning;;
                          *INFO*|info)  loglevel=info;;
            14013: |2
                      case "${__OCF_PRIO}" in
                        crit) __OCF_PRIO="CRIT";;
                        err)  __OCF_PRIO="ERROR";;
                        warn) __OCF_PRIO="WARNING";;
                        info) __OCF_PRIO="INFO";;
            18984: |2

              #------------------------------------------------------------------------------
              # ERROR REPORTING AND LOGGING
              #------------------------------------------------------------------------------
            19055: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic
            19069: |2
                        #   error
                        #   log
                        #   fatal
                        #   panic (effectively off)
            19258: |2

              #------------------------------------------------------------------------------
              # ERROR HANDLING
              #------------------------------------------------------------------------------
            30084: |2

                if node['fqdn'].nil?
                  Chef::Log.fatal("This node's fqdn is set to nil, so a read token cannot be issued!" \
                                  "Please change your fqdn settings.")
                end
            31704: |
              \33[0mPG::StringDataRightTruncation\33[0m
              -----------------------------\33[0m
              ERROR:  value too long for type character(32)
              \33[0m
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `exec_params'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:133:in `simple_insert'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:102:in `create_superuser_in_erchef'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:50:in `block in bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:55:in `block in with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:44:in `with_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/ec_postgres.rb:52:in `with_service_connection'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:49:in `bootstrap'
              \33[0m/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:46:in `block (2 levels) in from_file'\33[0m

              \33[0mResource Declaration:\33[0m
            31815: |
              \33[0mCookbook Trace:\33[0m
              ---------------\33[0m
              /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:184:in `restart_service'\33[0m

              \33[0mResource Declaration:\33[0m
            31911: |
              [2016-04-29T01:28:31+00:00] INFO: Processing execute[restart_opscode-erchef_log_service] action run (private-chef::opscode-erchef line 19)
              [2016-04-29T01:28:32+00:00] INFO: execute[restart_opscode-erchef_log_service] ran successfully
              [2016-04-29T01:28:32+00:00] ERROR: Running exception handlers
              [2016-04-29T01:28:32+00:00] ERROR: Exception handlers complete
              [2016-04-29T01:28:32+00:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
              [2016-04-29T01:28:32+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
              [2016-04-29T01:28:33+00:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
              * PG::StringDataRightTruncation occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 44) had an error: PG::StringDataRightTruncation: ERROR:  value too long for type character(32)

              * Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
            31935: |
              ============================================================

              /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant/config.rb:34:in `from_argv': Configuration file '/var/opt/opscode/oc-chef-pedant/etc/pedant_config.rb' not found! (RuntimeError)
                from /opt/opscode/embedded/service/oc-chef-pedant/lib/pedant.rb:54:in `setup'
                from ./bin/oc-chef-pedant:10:in `<main>'
              Build step 'Invoke XShell command' marked build as failure
              Performing Post build task...
              Match found for : : True
            31959: |
              Recording fingerprints
              Recording test results
              ERROR: Publisher hudson.tasks.junit.JUnitResultArchiver aborted due to exception
              hudson.AbortException: No test report files were found. Configuration error?
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:116)
                at hudson.tasks.junit.JUnitParser$ParseResultCallable.invoke(JUnitParser.java:92)
                at hudson.FilePath$FileCallableWrapper.call(FilePath.java:2688)
                at hudson.remoting.UserRequest.perform(UserRequest.java:121)
                at hudson.remoting.UserRequest.perform(UserRequest.java:49)
                at hudson.remoting.Request$2.run(Request.java:325)
                at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
                at java.util.concurrent.FutureTask.run(FutureTask.java:262)
                at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
                at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
                at java.lang.Thread.run(Thread.java:745)
                at ......remote call to chef-server-ubuntu-14.04-tester-69fc04(Native Method)
                at hudson.remoting.Channel.attachCallSiteStackTrace(Channel.java:1360)
                at hudson.remoting.UserResponse.retrieve(UserRequest.java:221)
                at hudson.remoting.Channel.call(Channel.java:753)
                at hudson.FilePath.act(FilePath.java:978)
                at hudson.FilePath.act(FilePath.java:967)
                at hudson.tasks.junit.JUnitParser.parseResult(JUnitParser.java:89)
                at hudson.tasks.junit.JUnitResultArchiver.parse(JUnitResultArchiver.java:121)
                at hudson.tasks.junit.JUnitResultArchiver.perform(JUnitResultArchiver.java:138)
                at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:74)
                at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
                at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:761)
                at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:721)
                at hudson.model.Build$BuildExecution.post2(Build.java:183)
                at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:670)
                at hudson.model.Run.execute(Run.java:1766)
                at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
                at hudson.model.ResourceController.execute(ResourceController.java:98)
                at hudson.model.Executor.run(Executor.java:374)
              Regular expression run condition: Expression=[^(?!true)$], Label=[true]
              Run condition [Regular expression match] preventing perform for step [[Editable Email Notification]]
        steps:
          total: 7m6s
          before (setup time): 2m23s
          after (cleanup time): 4m39s
  chef-server-12-build:
    result: SUCCESS
    url: http://wilson.ci.chef.co/job/chef-server-12-build/250/
    duration: 1h11m20s
    runs:
      el-5:
        result: SUCCESS
        url: http://wilson.ci.chef.co/job/chef-server-12-build/architecture=x86_64,platform=el-5,project=chef-server,role=builder/250/
        duration: 1h9m
      el-6:
        result: SUCCESS
        url: http://wilson.ci.chef.co/job/chef-server-12-build/architecture=x86_64,platform=el-6,project=chef-server,role=builder/250/
        duration: 1h11m11s
      el-7:
        result: SUCCESS
        url: http://wilson.ci.chef.co/job/chef-server-12-build/architecture=x86_64,platform=el-7,project=chef-server,role=builder/250/
        duration: 56m41s
      ubuntu-10.04:
        result: SUCCESS
        url: http://wilson.ci.chef.co/job/chef-server-12-build/architecture=x86_64,platform=ubuntu-10.04,project=chef-server,role=builder/250/
        duration: 48m57s
  chef-server-12-trigger-ad_hoc:
    result: SUCCESS
    url: http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/149/
    duration: 8s
successful_logs_analyzed: false
